scheduler/scheduler-examples/src/main/java/org/ow2/proactive/scheduler/examples/SFTPConnector.java

```diff
@@ -0,0 +1,366 @@
+                    // Create remote directory(ies) if it(they) does(do) not exist
+                    if (!remoteDirExists(channelsftp, sftpRemoteRelativePath)) {
+                        makeRemoteDirectories(channelsftp, Paths.get(sftpRemoteRelativePath));
+                    }
+                    // recursive folder content upload to sftp server
+                    filesRelativePathName = recursiveFolderUpload(channelsftp,
+                                                                  sftpLocalRelativePath,
+                                                                  sftpRemoteRelativePath);
+                    getOut().println("END UPLOAD TO SFTP");
+                    break;
+
+                default:
+                    throw new IllegalArgumentException("SFTP mode can only be PUT or GET.");
+            }
+        } catch (JSchException e) {
+            throw new JSchException("An error occured while trying to connect to SFTP server: ", e);
+        } catch (SftpException e) {
+            throw new SftpException(e.id,
+                                    "File not found, permission denied or operation not supported by SFTP server.",
+                                    e);
+        } finally {
+            if (channelsftp != null)
+                channelsftp.disconnect();
+            if (channel != null)
+                channel.disconnect();
+            if (session != null)
+                session.disconnect();
+            getOut().println("Disconnected from SFTP server");
+        }
+
+        return (Serializable) filesRelativePathName;
+    }
+
+    /**
+     * This method recursively uploads a local file/folder to an SFTP server
+     *
+     * @param sourcePath
+     * @param destinationPath
+     * @throws SftpException
+     * @throws FileNotFoundException
+     */
+    private List<String> recursiveFolderUpload(ChannelSftp channelSftp, String sourcePath, String destinationPath)
+            throws SftpException, FileNotFoundException {
+
+        List<String> filesRelativePathName = new ArrayList<>();
+
+        File sourceFile = new File(sourcePath);
+        if (sourceFile.isFile()) {
+            // copy if it is a file
+            channelSftp.cd(destinationPath);
+            getOut().println("Uploading " + sourcePath);
+            channelSftp.put(new FileInputStream(sourceFile), sourceFile.getName(), ChannelSftp.OVERWRITE);
+            getOut().println("File uploaded successfully to " +
+                             Paths.get(destinationPath, sourceFile.getName()).toString());
+            filesRelativePathName.add(Paths.get(destinationPath, sourceFile.getName()).toString());
+
+        } else {
+
+            String remoteFilePath = Paths.get(destinationPath, sourceFile.getName()).toString();
+            File[] files = sourceFile.listFiles();
+
+            if (files != null && !sourceFile.getName().startsWith(".")) {
```

Comment:
maybe consider isHidden() ?

Reply:
Yes, besides, that would also work for Windows-hidden files that are not prefixed with '.' (in case of sftp server on windows using OpenSSH). 
Thank you @fviale